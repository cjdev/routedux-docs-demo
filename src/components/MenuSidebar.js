import React from 'react';
import { SideNav, Link as SideNavLink } from '@cjdev/visual-stack/lib/components/SideNav';
import { connect } from 'react-redux';

function MenuSidebar({items, changePage}) {
  return (
    <SideNav>
      {items.map((item) => {
        return (
          <SideNavLink key={item.id}>
            <a onClick={
                 (ev) => {
                   ev.preventDefault();
                   changePage(item.id);
                 }
              }>{item.title}</a>
          </SideNavLink>
        );
      })}
    </SideNav>
  );
};

let mapDispatchToProps = (dispatch) => ({
  changePage(id) {
    dispatch({type: 'CHANGE_ID', id});
  }
});

export default connect(
  () => ({}),
  mapDispatchToProps
)(MenuSidebar);
