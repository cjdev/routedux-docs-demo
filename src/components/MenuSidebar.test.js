'use strict';

import React from 'react';
import ezJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import MenuSidebar from './MenuSidebar';


describe("menu does stuff you expect", () => {
  it("renders a list of article info into ActionLink components", () => {
    // given
    const data = [
      {id: "woo", title: "The woo", url: "woo"},
      {id: "bar", title: "The bar", url: "bar"}
    ];

    // when
    const rendered = shallow(<MenuSidebar items={data} routeBase="/articleroute/" />)

    // then
    expect(ezJson(rendered)).toMatchSnapshot()
  });
});