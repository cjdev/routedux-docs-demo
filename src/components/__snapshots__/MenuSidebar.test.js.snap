// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu does stuff you expect renders a list of article info into a link 1`] = `
<SideNav>
  <Link>
    <a
      onClick={[Function]}
    >
      The woo
    </a>
  </Link>
  <Link>
    <a
      onClick={[Function]}
    >
      The bar
    </a>
  </Link>
</SideNav>
`;
